Fix a number of runtimes

neersighted opened the pull request:
See commits

Fixes #15267

neersighted committed:
Fix canister runtime

The following runtime has occured 10198 time(s).
runtime error: Cannot execute null.return pressure().
proc name: update icon (/obj/machinery/portable_atmospherics/canister/update_icon)
  source file: canister.dm,112
  usr: Slurpy Slurpy (/mob/living/carbon/human)
  src: the o2 canister (/obj/machinery/portable_atmospherics/canister/oxygen)

neersighted committed:
Fix IsAdminGhost runtime

The following runtime has occured 263 time(s).
runtime error: Cannot read null.client
proc name: IsAdminGhost (/proc/IsAdminGhost)
  source file: mob_helpers.dm,398
  usr: Nanotrasen Private Security Of... (/mob/living/simple_animal/hostile/nanotrasen/ranged/smg)
  src: null

neersighted committed:
Fix sleeper runtime

Doesn't matter because the parent calls dropContents()

The following runtime has occured 128 time(s).
runtime error: Cannot read null.contents
proc name: Destroy (/obj/machinery/sleeper/Destroy)
  source file: Sleeper.dm,52
  usr: Daniel Young (/mob/living/carbon/human)
  src: the sleeper (/obj/machinery/sleeper)

neersighted committed:
Fix synth runtime

Synth damage overlays only apply if disguised

The following runtime has occured 80 time(s).
runtime error: Cannot read null.appearance
proc name: handle disguise (/datum/species/synth/proc/handle_disguise)
  source file: species_types.dm,649
  usr: null
  src: Military Synth (/datum/species/synth/military)

neersighted committed:
Use more reliable method of relabeling canister

neersighted committed:
Fix objective runtime

When reading fromg gases you must assert_gas

The following runtime has occured 59 time(s).
runtime error: cannot read from list
proc name: check special completion (/datum/objective_item/steal/plasma/check_special_completion)
  source file: objective_items.dm,97
  usr: null
  src: 28 moles of plasma (full tank) (/datum/objective_item/steal/plasma)

neersighted committed:
Fix gravity generator runtime

Checking turf Z is silly when mob checks make more sense

The following runtime has occured 53 time(s).
runtime error: Cannot read null.z
proc name: shake everyone (/obj/machinery/gravity_generator/main/proc/shake_everyone)
  source file: gravitygenerator.dm,360
  usr: null
  src: the gravitational generator (/obj/machinery/gravity_generator/main/station)

neersighted committed:
Fix click catcher runtime

Check that a turf exists when attempting to click it
This can happen if someone manages to click off the edge of the Z-level

The following runtime has occured 48 time(s).
runtime error: Cannot execute null.Click().
proc name: Click (/obj/screen/click_catcher/Click)
  source file: click.dm,345
  usr: Drone (758) (/mob/living/simple_animal/drone)
  src:  (/obj/screen/click_catcher)

neersighted committed:
istypeless meme

KorPhaeron commented on code/game/machinery/Sleeper.dm:
```diff
@@ -47,11 +47,6 @@
-	var/turf/T = loc
-	T.contents += contents
-	return ..()
-
```
Does this mean people get deleted on Destroy again?

neersighted replied:
Read the commit message please. All machines `dropContents()` on destroy.

neersighted committed:
Fix gravity gen logging during init

neersighted committed:
Fix an air alarm runtime

Don't store hard refs to an area which can cause race condtions
Instead, get our area only when we need it

The following runtime has occured 4426 time(s).
runtime error: Cannot read null.atmosalm
proc name: update icon (/obj/machinery/airalarm/update_icon)
  source file: airalarm.dm,535
  usr: null
  src: the alarm (/obj/machinery/airalarm)

neersighted committed:
Fix storage runtime

Mobs without clients can't see into storage! They have no screen!

The following runtime has occured 92 time(s).
runtime error: Cannot read null.screen
proc name: show to (/obj/item/weapon/storage/proc/show_to)
  source file: storage.dm,104
  usr: Leo Pfeifer (/mob/living/carbon/human)
  src: the backpack (/obj/item/weapon/storage/backpack)

neersighted committed:
Fix DNA runtime

When an admin clicks cancel on change species, a null species
it set. Check for this in the set_species proc

The following runtime has occured 34 time(s).
runtime error: Cannot create objects of type null.
proc name: set species (/mob/living/carbon/set_species)
  source file: dna.dm,180
  usr: Phoebe Lotsu (/mob/dead/observer)
  src: Swae Lee (/mob/living/carbon/human)

neersighted committed:
Address feedback

Canisters mixtures can no longer be null thanks to accepting
one in New()

Just forget assert_gas in objectives
